Controllers/ApplicationController.cs

```diff
@@ -36,7 +42,7 @@ public ActionResult Index()
         /// </summary>
         /// <returns>RedirectResult</returns>
         [HttpGet]
-        public RedirectResult RequestBalances()
+        public async Task<RedirectResult> RequestBalances()
```

Comment:
I believe if you could already return a Task from upstream(which is the case here), the `await` and `asyc` are unnecessary. Same for other places.